test/e2e/snaps/test-snap-managestate.spec.js

```diff
@@ -8,7 +8,148 @@ const FixtureBuilder = require('../fixture-builder');
 const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
 
 describe('Test Snap manageState', function () {
-  it('can pop up manageState snap and do update get and clear', async function () {
+  it('can use the new state API', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        // navigate to test snaps page, then fill in the snapId
+        await driver.driver.get(TEST_SNAPS_WEBSITE_URL);
+
+        // wait for page to load
+        await driver.waitForSelector({
+          text: 'Installed Snaps',
+          tag: 'h2',
+        });
+
+        // scroll to manage-state snap
+        const snapButton1 = await driver.findElement('#connectstate');
+        await driver.scrollToElement(snapButton1);
+
+        // added delay for firefox (deflake)
+        await driver.delayFirefox(1000);
+
+        // wait for and click connect
+        await driver.waitForSelector('#connectstate');
+        await driver.clickElement('#connectstate');
+
+        // switch to metamask extension
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        // wait for and click connect
+        await driver.waitForSelector({
+          text: 'Connect',
+          tag: 'button',
+        });
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        // wait for and click confirm
+        await driver.waitForSelector({ text: 'Confirm' });
+        await driver.clickElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
+
+        // wait for and click ok and wait for window to close
+        await driver.waitForSelector({ text: 'OK' });
+        await driver.clickElementAndWaitForWindowToClose({
+          text: 'OK',
+          tag: 'button',
+        });
+
+        // fill and click send inputs on test snap page
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestSnaps);
+
+        // wait for npm installation success
+        await driver.waitForSelector({
+          css: '#connectstate',
+          text: 'Reconnect to State Snap',
+        });
+
+        // Enter data and click set
+        await driver.pasteIntoField('#setStateKey', 'foo');
+        await driver.pasteIntoField('#dataState', '"bar"');
+        const sendEncrypted = await driver.findElement('#sendState');
+        await driver.scrollToElement(sendEncrypted);
+        await driver.clickElement('#sendState');
+
+        // Check that the entire state blob was updated
+        await driver.waitForSelector({
+          css: '#encryptedStateResult',
+          text: JSON.stringify({ foo: 'bar' }, null, 2),
+        });
+
+        // Check that we can retrive one state key
+        await driver.pasteIntoField('#getState', 'foo');
```

Comment:
I noticed that when executing it locally against the firefox browser there was a failure as mentioned in the screenshot. Also, firefox browser is executed only after the PR is merged and hence you did see the test failure. Could please execute locally and check using the below commands 

```
Make sure you run only this 'can use the new state API' test in the file test-snap-managestate.spec.js
yarn build:test:flask:mv2
ENABLE_MV3=false yarn test:e2e:single test/e2e/snaps/test-snap-managestate.spec.js --browser=firefox
```

or you could [comment this line ](https://github.com/MetaMask/metamask-extension/blob/main/.circleci/config.yml#L177)and push up the changes then the PR will be executed against firefox browser and please remember to revert it back.
<img width="1322" alt="Screenshot 2025-01-15 at 1 32 23 PM" src="https://github.com/user-attachments/assets/38486705-29ee-4e41-9315-1a665d0affa1" />

I tried by including the scroll and then pasting the value but it did not work.
This below code worked for me locally 

```suggestion
        await driver.fill('#getState', 'foo');
```

Reply:
Fixed!